src/rollup/index.ts

```diff
@@ -238,6 +238,12 @@ export default function rollup(
 								code += `//# ${SOURCEMAPPING_URL}=${url}\n`;
 							}
 
+							const wasmModule = modules.find(x => extname(x.id) === '.wasm');
+
+							if (wasmModule !== undefined) {
+								promises.push(writeFile('./dist/addTwo.wasm', wasmModule.code));
+							}
```

Comment:
I have no idea how to emit the file correct since it's a single file build.

Reply:
This is why I was wondering if it's worth considering each chunk (file) as having a WASM and JS component, where if the WASM or JS component is empty it is just a WASM or JS file and not both. So Chunk as consisting of both WASM and JS modules, where the render process for a chunk then returns both a magic string and a buffer, and if either is empty it's a WASM or JS file. But perhaps that is quite odd.

The alternative would be a more generic emission API, or to just do a specific check for a WASM output.

Reply:
I would prefer a more generic emission API. I feel like it's more future proof and we should take advantage of the Graph (to handle spllited or nested wasm builds).

Note that I have to MagicString in the initial JS module and write the wasm module in a file.